src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -176,6 +185,32 @@ void BindingManager::HandleDeviceConnectionFailure(PeerId peerId, CHIP_ERROR err
 {
     // Simply release the entry, the connection will be re-established as needed.
     ChipLogError(AppServer, "Failed to establish connection to node 0x" ChipLogFormatX64, ChipLogValueX64(peerId.GetNodeId()));
+
+    FabricIndex fabricToRemove = kUndefinedFabricIndex;
+    NodeId nodeToRemove        = kUndefinedNodeId;
+
+    for (PendingNotificationEntry pendingNotification : mPendingNotificationMap)
+    {
+        EmberBindingTableEntry entry = BindingTable::GetInstance().GetAt(pendingNotification.mBindingEntryId);
+
+        PeerId peer = PeerIdForNode(mInitParams.mFabricTable, entry.fabricIndex, entry.nodeId);
+
+        if (peerId == peer)
+        {
+            fabricToRemove = entry.fabricIndex;
+            nodeToRemove   = entry.nodeId;
+
+            BindingManagerContext * context = static_cast<BindingManagerContext *>(pendingNotification.mContext);
+
+            context->DecrementConsumersNumber();
+            if (context->GetConsumersNumber() == 0)
+            {
+                mBoundDeviceContextReleaseHandler(context);
+            }
+        }
+    }
+    mPendingNotificationMap.RemoveAllEntriesForNode(fabricToRemove, nodeToRemove);
```

Comment:
I think we were keeping the pending bits here on purpose in case we manage to connect to the device later... @gjc13 thoughts on doing this?

Reply:
Hmm, I'm actually not sure what was the purpose. Would you suggest not releasing the context and not removing notification in case of connection failure?

Reply:
I think so, yes.  Or at least making that change in a separate PR to minimize risk.

Reply:
Ok, removed.